packages/next-yak/loaders/lib/transpileCssProp.ts

```
@@ -0,0 +1,127 @@
+import type { NodePath, types as babelTypes } from "@babel/core";
+import type { JSXAttribute, JSXElement } from "@babel/types";
+
+export const transpileCssProp = (
+  t: typeof babelTypes,
+  path: NodePath<JSXElement>,
+  runtimeInternalHelpers: Set<string>,
+) => {
+  const openingElement = path.node.openingElement;
+  const cssPropIndex = openingElement.attributes.findIndex(
+    (prop) =>
+      t.isJSXAttribute(prop) &&
+      t.isJSXIdentifier(prop.name) &&
+      prop.name.name === "css",
+  );
+  // if the css prop is not present, we don't need to do anything
+  if (cssPropIndex === -1) {
+    return;
+  }
+
+  // can only be a JSXAttribute, as we checked above
+  const cssPropValue = (openingElement.attributes[cssPropIndex] as JSXAttribute)
+    .value;
+
+  // if the css prop is not an expression, we don't need to do anything
+  // e.g. <div css="..." /> instead of <div css={css`...`} />
+  if (!t.isJSXExpressionContainer(cssPropValue)) {
+    return;
```

should we throw an error here?

---

I tried to create a good error description. Wdyt?